### DISCUSSION TITLE
Show info at startup if the auto-index page is used

### DISCUSSION THREAD

**Discussion by rodja** [zauberzeug/nicegui#1065] [Ideas / Feature Requests]
Thanks to the [auto-index page](https://nicegui.io/documentation#auto-index_page), NiceGUI is very easy to start with. It also helps when trying or sharing short code snippets. But [as correctly stated on Discord](https://discord.com/channels/1089836369431498784/1121137650011746385/1121485650424762440), in 99% of all more complex apps, pages rather be [generated for each user individually](https://nicegui.io/documentation#page).

Maybe we could show an info message on startup in addition to "NiceGUI is ready to go on ..." if the main route `/` is serving the auto-index page? Something like

    Note: all users will see the same state on '/'. Use https://nicegui.io/documentation/page to create individual versions.

**Comment by ksandvik** [comment]
Yes, either that or maybe even provide a unique / route for all apps as it's very seldom you want a 'shared' experience of a web instance.

**Reply by rodja** [reply]
> provide a unique / route for all apps

That is technically not possible, because we do not want to re-evaluate the main script. We need some kind of function (eg. the page decorator) to be called for each new client.

**Reply by thatschatt** [reply]
I'm currently using Nicegui to make control interfaces for lab equipment, where the shared state is a huge selling point (since there is only one instrument that the interface is hooked up to). So definitely we should keep the shared page feature, although I agree that a warning for folk who don't expect it is a good idea.

In fact, it would be nice to be able to specify more pages as being shared, for example to make a shared device settings page. So maybe the best would be to specify in the page decorator if a given page should be private or shared. If no decorator is given for the '/' page, then we could show a warning on startup that it is in shared mode, and suggest that an explicit decorator is used to suppress this warning?
